### ISSUE TITLE
improve the detection model

### ISSUE THREAD

**Issue by xiaohai12** [facebookresearch/maskrcnn-benchmark#1064]
## ❓ Questions and Help

I met a problem that my model will detect some background which looks like our target object such as same color as target object, do you have any idea how to solve this problem?

**Comment by Darshan2701** [comment]
@xiaohai12 The only way I see is to increase the dataset with different examples and try using data augmentation techniques? Also, did u train from scratch or used pre-trained models?

**Comment by xiaohai12** [comment]
> @xiaohai12 The only way I see is to increase the dataset with different examples and try using data augmentation techniques? Also, did u train from scratch or used pre-trained models?

Yes, I used the transfer-learning

**Comment by Darshan2701** [comment]
Seems like the model has not generalized well enough and hence you have false positives, did you try adding few more examples? Also, try training from scratch once and compare the results. They are indeed many ways in which a model MAY improvize. Following are such 
1. Increase the training set.
2. use data augmentation techniques (Vertical flipping, Horizontal flip, color jittering )
3. Fune tuning by varying the learning rate.
4. Checking or visualizing the activate maps to get an idea about what model is seeing.
5. Changing the SGD momentum optimizer to RMSProp.
6. Increasing the POST_NMS_TEST number.
A lot of experimentation needs to be done.
You can also automate a lot of these work by using Microsoft's Azure cloud called HyperDrive which selects the best hyper params based on some heuristics like Grid Search, Random Search, and Bayesian Optimization, etc.

**Comment by xiaohai12** [comment]
> Seems like the model has not generalized well enough and hence you have false positives, did you try adding few more examples? Also, try training from scratch once and compare the results. They are indeed many ways in which a model MAY improvize. Following are such
> 
> 1. Increase the training set.
> 2. use data augmentation techniques (Vertical flipping, Horizontal flip, color jittering )
> 3. Fune tuning by varying the learning rate.
> 4. Checking or visualizing the activate maps to get an idea about what model is seeing.
> 5. Changing the SGD momentum optimizer to RMSProp.
> 6. Increasing the POST_NMS_TEST number.
>    A lot of experimentation needs to be done.
>    You can also automate a lot of these work by using Microsoft's Azure cloud called HyperDrive which selects the best hyper params based on some heuristics like Grid Search, Random Search, and Bayesian Optimization, etc.

Thanks for your advice.
Today is my last day working on it,  I will try to some of your advice, and tell you if it works.
We don't want to increase training data and I already tried data augmentation(useless).

**Comment by Darshan2701** [comment]
@xiaohai12 Yes, please. Data Augmentation using mere color jittering did not work for me also. Maybe we need more advanced techniques like random scaling or random cropping as in Keras.
